python/cudf/cudf/core/index.py

```diff
@@ -3516,10 +3522,15 @@ def __init__(
         if yearfirst is not False:
             raise NotImplementedError("yearfirst == True is not yet supported")
 
-        if freq is None and isinstance(data, type(self)):
-            freq = data.freq
+        if freq is None:
+            if isinstance(data, type(self)):
+                freq = data.freq
+            if isinstance(data, pd.DatetimeIndex) and data.freq is not None:
+                freq = data.freq.freqstr
 
         name = _getdefault_name(data, name=name)
+
+        was_pd_index = isinstance(data, pd.DatetimeIndex)
```

Comment:
Move this above line 3525 and you can use this variable to remove an extra instance check in there.

Reply:
This is redundant now
```suggestion
```